File: examples/platforms/android/qtquickview_java/app/src/main/java/com/example/qtquickview_java/Colors.java
// Copyright (C) 2024 The Qt Company Ltd.
// SPDX-License-Identifier: LicenseRef-Qt-Commercial OR BSD-3-Clause
package com.example.qtquickview_java;

import java.util.Arrays;
import java.util.Collections;
import java.util.Stack;

class Colors {
    private final Stack<Integer> recycle;
    private final Stack<Integer> colors;

    public Colors() {
        colors = new Stack<>();
        recycle = new Stack<>();
        recycle.addAll(Arrays.asList(
                        0xff1CB669, 0xff00414A, 0xff27138B,
                        0xffB5C10E, 0xff373F26, 0xffAF93DF,
                        0xff817505
                )
        );
    }

    public String getColor() {
        if (colors.size()==0) {
            while (!recycle.isEmpty())
                colors.push(recycle.pop());
            Collections.shuffle(colors);
        }
        int color = colors.pop();
        recycle.push(color);
        return String.format("#%06X", (0xFFFFFF & color));
    }
}

File: examples/platforms/android/qtquickview_kotlin/app/src/main/java/com/example/qtquickview_kotlin/Colors.kt
// Copyright (C) 2024 The Qt Company Ltd.
// SPDX-License-Identifier: LicenseRef-Qt-Commercial OR BSD-3-Clause
package com.example.qtquickview_kotlin

import java.util.Collections
import java.util.Stack

internal class Colors {
    private val recycle: Stack<Int> = Stack()
    private val colors: Stack<Int> = Stack()

    init {
        recycle.addAll(
            mutableListOf(
                -0xe34997, -0xffbeb6, -0xd8ec75,
                -0x4a3ef2, -0xc8c0da, -0x506c21,
                -0x7e8afb
            )
        )
    }

    fun getColor(): String {
            if (colors.size == 0) {
                while (!recycle.isEmpty()) colors.push(recycle.pop())
                Collections.shuffle(colors)
            }
            val color = colors.pop()
            recycle.push(color)
            return String.format("#%06X", 0xFFFFFF and color)
        }
}
